ee/codegen/src/generators/nodes/search-node.ts

```diff
@@ -192,19 +192,27 @@ export class SearchNode extends BaseSingleFileNode<
         rule.type === "CONSTANT_VALUE" && rule.data.type === "JSON"
     );
 
-    if (nodeInputValuePointer) {
-      const rawData = nodeInputValuePointer.data;
+    if (!nodeInputValuePointer) {
+      return;
+    }
 
-      const metadataFilter = rawData.value;
+    const metadataFilters = nodeInputValuePointer.data.value;
+    if (!metadataFilters) {
+      return undefined;
+    }
 
-      const parsedData =
-        VellumValueLogicalExpressionSerializer.parse(metadataFilter);
+    const parsedData =
+      VellumValueLogicalExpressionSerializer.parse(metadataFilters);
 
-      if (parsedData.ok) {
-        return parsedData.value;
-      }
+    if (!parsedData.ok) {
+      throw new Error(
+        `Failed to parse metadata filter JSON: ${JSON.stringify(
+          parsedData.errors
+        )}`
+      );
     }
-    return undefined;
```

Comment:
We were previously silently swallowing JSON parsing errors when we should instead fail loudly and early.